Source/Core/DolphinQt/Debugger/MemoryWidget.cpp

```diff
@@ -476,12 +495,123 @@ void MemoryWidget::OnSearchAddress()
   m_search_offset->setPalette(offset_palette);
 }
 
-void MemoryWidget::ValidateSearchValue()
+void MemoryWidget::ValidateAndPreviewInputValue()
 {
+  m_data_preview->clear();
+  QString input_text = m_data_edit->text();
+
+  if (input_text.isEmpty())
+    return;
+
+  // Remove any spaces
+  if (!m_input_ascii->isChecked())
+    input_text = input_text.replace(QStringLiteral(" "), QStringLiteral(""));
+
   QFont font;
   QPalette palette;
+  QString hex_string;
+  int padding = 8;
+  int hex = 16;
+  bool good = true;
+
+  if (m_input_ascii->isChecked())
+  {
+    // MemoryView uses Latin1.
+    QByteArray bytes = input_text.toLatin1();
+
+    // An unrestrained size is fine, but it needs to truncate in the preview.
+    if (bytes.size() > 8)
+    {
+      bytes.truncate(8);
+      hex_string = QString::fromLatin1(bytes.toHex()).append(QStringLiteral("...."));
+    }
+    else
+    {
+      hex_string = QString::fromLatin1(bytes.toHex());
+    }
+  }
+  else if (m_input_float->isChecked())
+  {
+    const float float_in = input_text.toFloat(&good);
+
+    if (good)
+    {
+      const u32 hex_out = Common::BitCast<u32>(float_in);
+      hex_string = QStringLiteral("%1").arg(hex_out, padding, hex, QLatin1Char('0')).toUpper();
+    }
+  }
+  else if (m_input_integer->isChecked())
+  {
+    if (input_text.startsWith(QStringLiteral("-")))
```

Comment:
```suggestion
    if (input_text.startsWith(QLatin1Char('-')))
```

Should work here if I'm not mistaken

Reply:
Yep, good spot